src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -109,6 +110,16 @@ public Task<ComponentRenderedText> PrerenderComponentAsync(Type componentType, P
             });
         }
 
+        public void SetCircuitUser(ClaimsPrincipal user)
+        {
+            var authenticationStateProvider = Services.GetService<AuthenticationStateProvider>() as IHostEnvironmentAuthenticationStateProvider;
```

Comment:
should we make this **fail hard** if the cast doesn't succeed? For instance if someone wanted to customize the auth state provider, they could easily miss this detail and go into the weeds. I don't have a strong feeling about which is better to do.

Reply:
The intention is for "receiving the user info from the host environment" to be an optional feature for authentication state providers. If you want to build a custom one, maybe you want it to do something with ASP.NET Core's normal server-side `User` (in which case, implement `IHostEnvironmentAuthenticationStateProvider`), or maybe you don't (in which case, don't).

As long as this is documented with other coverage for "building a custom authentication state provider" then I think people will be on track to succeed.